jetbrains-core/src/software/aws/toolkits/jetbrains/core/credentials/ExtensionPointCredentialsProviderRegistry.kt

```diff
@@ -33,20 +33,24 @@ class CredentialProviderFactoryEP : AbstractExtensionPointBean() {
 }
 
 class ExtensionPointCredentialsProviderRegistry : ToolkitCredentialsProviderRegistry {
-    override fun listFactories(): Collection<ToolkitCredentialsProviderFactory> = EXTENSION_POINT.extensions
-        .mapNotNull {
-            try {
-                it.getHandler()
-            } catch (e: Exception) {
-                LOG.error(e)
-                null
-            }
-        }
-        .map { it.createToolkitCredentialProviderFactory() }
+    override fun listFactories(): Collection<ToolkitCredentialsProviderFactory> = factories
 
     companion object {
         private val LOG = Logger.getInstance(ExtensionPointCredentialsProviderRegistry::class.java)
         private const val EP_NAME = "aws.toolkit.credentialProviderFactory"
-        val EXTENSION_POINT = ExtensionPointName.create<CredentialProviderFactoryEP>(EP_NAME)
+        private val EXTENSION_POINT = ExtensionPointName.create<CredentialProviderFactoryEP>(EP_NAME)
+
+        private val factories by lazy {
+            EXTENSION_POINT.extensions
+                .mapNotNull {
+                    try {
```

Comment:
`LOG.tryOrNull { it.getHandler() }`

Reply:
That is an extension on slf4j, not the IJ diagnostic Logger

Reply:
Guess using slf4j isnt a big deal here

Reply:
one just delegates to the other so it's much of a muchness.